src/main/java/uk/ac/ebi/embl/gff3tools/validation/builtin/AntiCodonValidation.java

```diff
@@ -0,0 +1,120 @@
+    private static final String INVALID_AMINO_ACID = "%s contains illegal amino acid.";
+    private static final String INVALID_AMINO_ACID_VALUE =
+            "Illegal amino acid \"%s\" should be changed to legal amino acid \"%s\"";
+    private static final String INVALID_LOCATION_RANGE = "%s location is invalid. Length must be within feature range.";
+    private static final String INVALID_LOCATION_VALUE = "Invalid %s location: start must be > 0 and less than end.";
+    private static final String INVALID_LOCATION_SPAN = "%s location span must be \"%s\"";
+
+    @ValidationMethod(rule = "ANTI_CODON_ATTRIBUTE", type = ValidationType.FEATURE)
+    public void validateAntiCodon(GFF3Feature feature, int line) throws ValidationException {
+        if (feature.hasAttribute(GFF3Attributes.ANTI_CODON)) {
+            validateCodonAttribute(
+                    feature, line, GFF3Attributes.ANTI_CODON, feature.getAttributeByName(GFF3Attributes.ANTI_CODON));
+        }
+    }
+
+    @ValidationMethod(rule = "TRANSL_EXCEPT_ATTRIBUTE", type = ValidationType.FEATURE)
+    public void validateTranslExcept(GFF3Feature feature, int line) throws ValidationException {
+        if (feature.hasAttribute(GFF3Attributes.TRANSL_EXCEPT)) {
+            validateCodonAttribute(
+                    feature,
+                    line,
+                    GFF3Attributes.TRANSL_EXCEPT,
+                    feature.getAttributeByName(GFF3Attributes.TRANSL_EXCEPT));
+        }
+    }
+
+    private void validateCodonAttribute(GFF3Feature feature, int line, String attribute, String value)
+            throws ValidationException {
+        AminoAcid aminoAcid;
+        String aminoAcidString;
+        CompoundLocation<Location> location;
+
+        try {
+            if (attribute.equalsIgnoreCase(GFF3Attributes.ANTI_CODON)) {
+                AnticodonQualifier q = new AnticodonQualifier(value);
+                aminoAcid = q.getAminoAcid();
+                aminoAcidString = q.getAminoAcidString();
+                location = q.getLocations();
+            } else if (attribute.equalsIgnoreCase(GFF3Attributes.TRANSL_EXCEPT)) {
+                TranslExceptQualifier q = new TranslExceptQualifier(value);
+                aminoAcid = q.getAminoAcid();
+                aminoAcidString = q.getAminoAcidString();
+                location = q.getLocations();
+            } else {
+                return;
+            }
+
+            long start = location.getMinPosition();
+            long end = location.getMaxPosition();
+            long length = location.getLength();
+
+            // Feature boundaries
+            long featureStart = feature.getStart();
+            long featureEnd = feature.getEnd();
+
+            // Amino acid
+            if (aminoAcid == null) {
+                throw new ValidationException(line, INVALID_AMINO_ACID.formatted(attribute));
+            }
+            // TODO: The below case needs to be thrown as Warning
```

Comment:
If this must be a warning, then it must be defined as an independent validation.